frontend/src/metabase/query_builder/components/NativeQueryEditor.jsx

```diff
@@ -138,7 +138,7 @@ class NativeQueryEditor extends Component {
 
     const editorElement = this.editor.current;
 
-    if (query.hasWritePermission()) {
+    if (query.hasWritePermission() && !readOnly) {
       this._editor.setReadOnly(false);
       editorElement.classList.remove("read-only");
     } else {
```

Comment:
Would appreciate someone with deeper expertise around the native query editor validating that. I haven't seen the editor's `readOnly` prop used and it kinda makes sense to disable `ace` as well